docs: Add release notes for Metals 1.5.3 (#7459)
scalameta/metals: website/blog/2025-05-12-strontium.md

</tbody>
</table>

For full details: [https://github.com/scalameta/metals/milestone/78?closed=1](https://github.com/scalameta/metals/milestone/78?closed=1)

Metals is a language server for Scala that works with VS Code, Vim, Emacs, Zed,
Helix and Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from contributors from the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

## Menu
- [MCP support](#mcp-support)
- [Best effort compilation](#best-effort-compilation)
- [New inlay hints](#new-inlay-hints)
  - [Named parameters](#named-parameters)
  - [By-name parameters](#by-name-parameters)
- [Remove invalid imports code action](#remove-invalid-imports-code-action)
- [Convert to named lambda parameters code action](#convert-to-named-lambda-parameters-code-action)
- [Support using directives in worksheets](#support-using-directives-in-worksheets)
- [Improve auto-fill arguments](#improve-auto-fill-arguments)
- [Discover tests for ZIO test framework](#discover-tests-for-zio-test-framework)
- [Miscellaneous](#miscellaneous)

## MCP support

> Starting with this release, Metals can serve an sse, MCP server.

Reviewer: ```suggestion
Starting with this release, Metals can serve an SSE MCP server.
```

Furthermore, giving the AI agent the capability to compile files, run tests, and import builds can further help automate your workflow.

### How to use Metals MCP?
To enable Metals MCP support, set `metals.startMcpServer` to `true`. For common editors like Cursor or VSCode (GitHub Copilot), MCP configuration will be automatically added to your workspace. However, you should still check in your AI agent settings that the connection was made successfully. For other AI, Metals will show a message upon MCP server startup with the port on which it was started locally. If you miss it, `Metals MCP server started on port: ${port}` message is also printed in the Metals log (`.metals/metals.log`).

### Current list of MCP tools:
- `compile-file` -- Compile a chosen Scala file.
- `compile-full` -- Compile the whole Scala project.
- `test` -- Run Scala test suite.
- `glob-search` -- Search for symbols containing substring

> - `typed-glob-search` -- Search for symbols containing a substring, that match the allowed symbol kinds (e.g., class, method).

Reviewer: ```suggestion
- `typed-glob-search` -- Search for symbols containing a substring that match the allowed symbol kinds (e.g., class, method).
```

## New inlay hints

### Named parameters
[tgodzik](https://github.com/tgodzik) added inlay hints for showing parameters' names on call site. The new inlay hint is available under `inlay-hints.named-parameters.enable` Metals setting, for now available only for Scala 2.
![named-params](https://i.imgur.com/MJwFuqs.png)
*Named parameters inlay hints*

### By-name parameters
We also have new inlay hints marking by-name parameters with `=>` on the call site. Thanks go to [harpocrates](https://github.com/harpocrates).
![by-name-params](https://i.imgur.com/sk2k6Ay.png)

## Remove invalid imports code action
This release also has a new code action for removing invalid imports, saving you the trouble of deleting them manually. It is yet another improvement added by [harpocrates](https://github.com/harpocrates). Code action works both for Scala 2 and Scala 3, though older Scala 2 versions might get only partial support.

![remove-invalid-imports](https://i.imgur.com/mArQW9N.gif)
*Remove invalid import*
![remove-invalid-imports-2](https://i.imgur.com/Ht9htWM.gif)
*Remove invalid import in multi-import statement*

## Convert to named lambda parameters code action

> Thanks to [KacperFKorban](https://github.com/KacperFKorban) we have a new code action that converts a wildcard lambda into a lambda with parameters. This was only implemented for Scala 3 and will work from versions `3.8.x` and LTS `3.3.8`.

Reviewer: ```suggestion
Thanks to [KacperFKorban](https://github.com/KacperFKorban) we have a new code action that converts a wildcard lambda into a lambda with parameters. This was only implemented for Scala 3 and will work from versions `3.7.2` and LTS `3.3.7`.
```
no?

Author: Yeah, I think, you're right.
